restapi/src/main/java/io/stargate/web/docsapi/service/DocumentService.java

```diff
@@ -390,6 +391,110 @@ private Object convertToBackendBooleanValue(boolean value, boolean numericBoolea
     return ImmutablePair.of(bindVariableList, firstLevelKeys);
   }
 
+  private String convertToJsonPtr(String path) {
+    return "/" + path.replaceAll(PERIOD_PATTERN.pattern(), "/").replaceAll("\\[(\\d+)\\]", "$1");
+  }
+
+  public List<String> writeManyDocs(
+      String authToken,
+      String keyspace,
+      String collection,
+      InputStream payload,
+      Optional<String> idPath,
+      Db dbFactory,
+      Map<String, String> headers)
+      throws IOException, UnauthorizedException {
+
+    DocumentDB db = dbFactory.getDocDataStoreForToken(authToken, headers);
+    JsonSurfer surfer = JsonSurferGson.INSTANCE;
+
+    boolean created = db.maybeCreateTable(keyspace, collection);
+    // After creating the table, it can take up to 2 seconds for permissions cache to be updated,
+    // but we can force the permissions refetch by logging in again.
+    if (created) {
+      db = dbFactory.getDocDataStoreForToken(authToken, headers);
+      db.maybeCreateTableIndexes(keyspace, collection);
+    }
+    List<String> idsWritten = new ArrayList<>();
+    try (BufferedReader reader = new BufferedReader(new InputStreamReader(payload, "UTF-8"))) {
+      Iterator<String> iter = reader.lines().iterator();
+      ExecutionContext context = ExecutionContext.NOOP_CONTEXT;
+      int chunkIndex = 0;
+      final int CHUNK_SIZE = 250;
+      while (iter.hasNext()) {
+        chunkIndex++;
+        Map<String, String> docsInChunk = new LinkedHashMap<>();
+        while (docsInChunk.size() < CHUNK_SIZE && iter.hasNext()) {
+          String doc = iter.next();
+          JsonNode json = mapper.readTree(doc);
+
+          String docId = UUID.randomUUID().toString();
+          if (idPath.isPresent()) {
+            String docsPath = convertToJsonPtr(idPath.get());
```

Comment:
so everything related to the `idPath` conversion to the doc path can be moved out of the loop, this way you are not doing the same thing over and over again as this seems to be a constant..